### DISCUSSION TITLE
hasRole in Gate return false with middleware

### DISCUSSION THREAD

**Discussion by dmitrydrynov** [spatie/laravel-permission#2109] [Q&A]
hasRole('superadmin') in Gate return always false with middleware.

web.php
```
Route::group(['middleware' => 'auth'], static function () {
    Route::get('', ['uses' => 'App\Http\Controllers\Member\IndexController@index', 'as' => 'index'])->middleware(['can:members.read']);
});
```

AuthServiceProvider.php
```
public function boot(): void
{
      ...

      $this->registerPolicies();

      Gate::before(function ($user) {
            return $user->hasRole('superadmin') ? true : null;
      });
}
```

If I use `$user->hasRole('role_name')` in code or `can('some_permission')` in view I have a correct result. 
When I use this in route's middleware hasRole function return false always.

p.s. My superadmin role doesn't have any permissions only role.

Help please :)

**Comment by erikn69** [comment]
Try `auth` middleware first

**Reply by dmitrydrynov** [reply]
auth middleware used already. I have Route::group with auth middleware (added in the first messsage)

**Reply by erikn69** [reply]
It's hard to answer without all the information, laravel version, php version, package version and others
Maybe you could make a failing test, or a minimal laravel application for bug search
